### PR TITLE
New version with better logs for porto isolation

### PR DESCRIPTION
**Pull request by dkarpukhin**

### CODE REVIEW

File: isolate/initialdispatch.go [hunk 1]
```diff
@@ -185,6 +185,8 @@ func (d *initialDispatch) onSpawn(opts Profile, name, executable string, args, e
 		return nil, err
 	}
 
+	apexctx.GetLogger(d.ctx).Debugf("onSpawn() Profile Dump: %s", opts.String())
```
Comment (noxiouz) [inline isolate/initialdispatch.go]: an explicit String() call avoids lazy evaluation. fmt calls it for %s

File: isolate/porto/container.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"strconv"
 	"syscall"
 	"time"
+	"fmt"
```
Comment (noxiouz) [inline isolate/porto/container.go]: alphabet order

File: isolate/porto/container.go [hunk 3]
```diff
@@ -80,6 +81,7 @@ func pickNetwork(network string) string {
 // NOTE: is it better to have some kind of our own init inside Porto container to handle output?
 
 func newContainer(ctx context.Context, portoConn porto.API, cfg containerConfig, info execInfo) (cnt *container, err error) {
+	apexctx.GetLogger(ctx).WithField("container", cfg.ID).Debugf("exec newContainer() with containerConfig: %s; execInfo: %s;", fmt.Sprintf("%s", cfg), fmt.Sprintf("%s", info))
```
Comment (noxiouz) [inline isolate/porto/container.go]: drop fmt.Sprintf

File: isolate/profile.go [hunk 4]
```diff
@@ -2,10 +2,21 @@ package isolate
...
 type Profile map[string]interface{}
 
 func (p Profile) Type() string {
 	return fmt.Sprintf("%s", p["type"])
 }
+
+func (p Profile) String() string {
+	j, e := json.Marshal(p)
+	if e == nil {
+		return fmt.Sprintf("%s", j)
```
Comment (noxiouz) [inline isolate/profile.go]:
```go
return string(j)
```

File: isolate/profile.go [hunk 5]
```diff
@@ -2,10 +2,21 @@ package isolate
...
+	} else {
```
Comment (noxiouz) [inline isolate/profile.go]: drop else, just return "nil"

**Comment by dkarpukhin** [comment]
fixed

**Comment by noxiouz** [comment]
Close in favor to #111

**Pull request closed without merging**